### PR TITLE
fix: Fix some issues with existing code

### PR DESCRIPTION
**Pull request by denisonbarbosa**
Some tests that were not marked to fail were failing, so I took a look to understand why. Details for each fix are explained in the commit messages.

### CODE REVIEW

File: cmd/oidc-broker/daemon/daemon.go [hunk 1]
```diff
@@ -44,31 +45,35 @@ type daemonConfig struct {
 }
 
 // New registers commands and return a new App.
-func New() *App {
-	a := App{ready: make(chan struct{})}
+func New(name string) *App {
+	if name == "" {
```
Comment (didrocks) [inline cmd/oidc-broker/daemon/daemon.go:49]: I would directly change to not make it optional then, and remove the global name altogether (to ensure we don’t use it by accident)

File: internal/brokerservice/brokerservice.go [hunk 2]
```diff
@@ -107,7 +108,12 @@ func (s *Service) Serve() error {
 
 // Stop stop the service and do all the necessary cleanup operation.
 func (s *Service) Stop() error {
-	close(s.serve)
-	s.disconnect()
+	// Check if already stopped.
+	select {
+	case <-s.serve:
+	case <-time.After(10 * time.Millisecond):
+		close(s.serve)
```
Comment (didrocks) [inline internal/brokerservice/brokerservice.go:115]: unsure how this fixes the race/double close case? was it something we were doing in adsys or authd?

**Review by didrocks — CHANGES_REQUESTED** [review]
One change requested and one question!

File: internal/brokerservice/brokerservice.go (hunk 2 above)
Comment (denisonbarbosa) [reply internal/brokerservice/brokerservice.go:115]: Receiving from a closed channel is a non-blocking no-op. If the channel is not closed, it will hang there the same way as the `serve()` function does and then we close it after a little delay.

TLDR: If the channel is already closed, Stop() is a no-op.

Comment (didrocks) [reply internal/brokerservice/brokerservice.go:115]: instead of a small delay (which introduce a bigger race window if we don’t use mutex), why not use `default:`?

Comment (denisonbarbosa) [reply internal/brokerservice/brokerservice.go:115]: You are very smart, mister. I didn't think of it at all 🤣

File: cmd/oidc-broker/main.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ import (
 
 func main() {
 	i18n.InitI18nDomain(consts.TEXTDOMAIN, po.Files)
-	a := daemon.New("")
+	a := daemon.New("oidc-broker")
```
Comment (didrocks) [inline cmd/oidc-broker/main.go:23]: No, because we will not be able to rename it. This is why (see the diff, it’s `filepath.Base(os.Args[0])`) and not hardcoded.

Comment (denisonbarbosa) [reply cmd/oidc-broker/main.go:23]: oops, you're right! I'll fix it.

**Review by didrocks — APPROVED** [review]
All good!
